Stratis.Bitcoin.Features.Consensus/StakeItem.cs

```diff
@@ -1,13 +1,84 @@
 ﻿using NBitcoin;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.Features.Consensus
 {
-    public class StakeItem
+    /// <summary>
+    /// An object that holds block stake information.
+    /// </summary>
+    public sealed class StakeItem
     {
-        public uint256 BlockId;
-        public BlockStake BlockStake;
-        public bool InStore;
-        public long Height;
-    }
-}
+        private StakeItem() { }
+
+        /// <summary>The hash of the block.</summary>
+        public uint256 BlockHash { get; private set; }
+
+        /// <summary>The block stake.</summary>
```

Comment:
I think this is rather `staking information`, "block stake" is a name of the class or variable.